doc/modules/partial_dependence.rst

```diff
@@ -89,14 +99,72 @@ The values at which the partial dependence should be evaluated are directly
 generated from ``X``. For 2-way partial dependence, a 2D-grid of values is
 generated. The ``values`` field returned by
 :func:`sklearn.inspection.partial_dependence` gives the actual values
-used in the grid for each target feature. They also correspond to the axis
-of the plots.
+used in the grid for each input feature of interest. They also correspond to
+the axis of the plots.
+
+Individual conditional expectation (ICE) plot
+=============================================
+
+Similar to a PDP, an individual conditional expectation (ICE) plot
+shows the dependence between the target function and an input feature of
+interest. However, unlike a PDP, which shows the average effect of the input
+feature, an ICE plot visualizes the dependence of the prediction on a
+feature for each sample separately with one line per sample.
+Due to the limits of human perception, only one input feature of interest is
+supported for ICE plots.
+
+The figures below show four ICE plots for the California housing dataset,
+with a :class:`GradientBoostingRegressor
+<sklearn.ensemble.GradientBoostingRegressor>`. The second figure plots
+the corresponding PD line overlaid on ICE lines.
+
+.. figure:: ../auto_examples/inspection/images/sphx_glr_plot_partial_dependence_004.png
+   :target: ../auto_examples/inspection/plot_partial_dependence.html
+   :align: center
+   :scale: 70
+
+.. figure:: ../auto_examples/inspection/images/sphx_glr_plot_partial_dependence_005.png
+   :target: ../auto_examples/inspection/plot_partial_dependence.html
+   :align: center
+   :scale: 70
+
+While the PDPs are good at showing the average effect of the target features,
+they can obscure a heterogeneous relationship created by interactions.
+When interactions are present the ICE plot will provide many more insights.
+For example, we could observe a linear relationship between the median income
+and the house price in the PD line. However, the ICE lines show that there
+are some exceptions, where the house price remains constant in some ranges of
+the median income.
+ 
+The :mod:`sklearn.inspection` module's :func:`plot_partial_dependence`
+convenience function can be used to create ICE plots by setting
+``individual=True``. In the example below, we show how to create a grid of
+ICE plots:
+
+    >>> from sklearn.datasets import make_hastie_10_2
```

Comment:
I would remove the example and rather link the real example

Reply:
@glemaitre Do you mean linking to https://97096-843222-gh.circle-artifacts.com/0/doc/auto_examples/inspection/plot_partial_dependence.html? It is anyway linked at the end of the page.

Reply:
Oh sorry I thought I removed this comment. Keep the example since we have one in the above section in a similar manner. At first, I thought we did not have one above.

Reply:
No problem. I'll mark this as resolved.